hammer/workerpool.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import "context"
+
+type worker interface {
+	Run(ctx context.Context)
+	Kill()
+}
+
+func newWorkerPool(factory func() worker) workerPool {
```

Comment:
Came across this recently, dunno if it's useful here or overkill: https://github.com/alitto/pond
(Not asking you to use it though, just pointing at it in case you want to.)

Reply:
Added a comment.